pcl_ros/CMakeLists.txt

```diff
@@ -1,245 +1,195 @@
-class_loader_hide_library_symbols(pcl_ros_features)
-
-
-## Declare the pcl_ros_filters library
-add_library(pcl_ros_filters
-  src/pcl_ros/filters/extract_indices.cpp
-  src/pcl_ros/filters/filter.cpp
-  src/pcl_ros/filters/passthrough.cpp
-  src/pcl_ros/filters/project_inliers.cpp
-  src/pcl_ros/filters/radius_outlier_removal.cpp
-  src/pcl_ros/filters/statistical_outlier_removal.cpp
-  src/pcl_ros/filters/voxel_grid.cpp
-  src/pcl_ros/filters/crop_box.cpp
-)
-target_link_libraries(pcl_ros_filters pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(pcl_ros_filters ${PROJECT_NAME}_gencfg)
-class_loader_hide_library_symbols(pcl_ros_filters)
-
-## Declare the pcl_ros_segmentation library
-add_library (pcl_ros_segmentation
-  src/pcl_ros/segmentation/extract_clusters.cpp
-  src/pcl_ros/segmentation/extract_polygonal_prism_data.cpp
-  src/pcl_ros/segmentation/sac_segmentation.cpp
-  src/pcl_ros/segmentation/segment_differences.cpp
-  src/pcl_ros/segmentation/segmentation.cpp
-)
-target_link_libraries(pcl_ros_segmentation pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(pcl_ros_segmentation ${PROJECT_NAME}_gencfg)
-class_loader_hide_library_symbols(pcl_ros_segmentation)
-
-## Declare the pcl_ros_surface library
-add_library (pcl_ros_surface
-  src/pcl_ros/surface/surface.cpp
-  # Compilation is much faster if we include all the following CPP files in surface.cpp
-  src/pcl_ros/surface/convex_hull.cpp
-  src/pcl_ros/surface/moving_least_squares.cpp
-)
-target_link_libraries(pcl_ros_surface ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(pcl_ros_surface ${PROJECT_NAME}_gencfg)
-class_loader_hide_library_symbols(pcl_ros_surface)
-
-## Tools
-
-add_executable(pcd_to_pointcloud tools/pcd_to_pointcloud.cpp)
-target_link_libraries(pcd_to_pointcloud ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(pointcloud_to_pcd tools/pointcloud_to_pcd.cpp)
-target_link_libraries(pointcloud_to_pcd ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(bag_to_pcd tools/bag_to_pcd.cpp)
-target_link_libraries(bag_to_pcd pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(convert_pcd_to_image tools/convert_pcd_to_image.cpp)
-target_link_libraries(convert_pcd_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(convert_pointcloud_to_image tools/convert_pointcloud_to_image.cpp)
-target_link_libraries(convert_pointcloud_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-## Downloads
-
-catkin_download(table_scene_lms400.pcd http://download.ros.org/data/pcl/table_scene_lms400.pcd
-  DESTINATION ${CMAKE_CURRENT_SOURCE_DIR}/samples/data
-  MD5 546b5b4822fb1de21b0cf83d41ad6683
-)
-add_custom_target(download ALL DEPENDS download_extra_data)
+target_include_directories(pcl_ros_tf
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
```

Comment:
Curious what's with this, just doing `include/` should work OK, what's the rest of this for?

Reply:
It's the recommended way to do modern CMake, at least as far as I'm aware. Deals with target exporting/importing stuff.

Maybe not a concern for ROS packages, am totally fine with old-style too:
```
target_include_directories(pcl_ros_tf include)
```

Reply:
done